src/main/java/org/jenkinsci/plugins/mesos/MesosApi.java

```diff
@@ -118,24 +90,20 @@ public MesosApi(String masterUrl, URL jenkinsUrl, String user, String frameworkN
    * @param materializer The {@link ActorMaterializer} used for the source queue.
    * @return A running source queue.
    */
-  private SourceQueueWithComplete<SpecUpdated> runUsi(
+  private CompletableFuture<SourceQueueWithComplete<SpecUpdated>> runUsi(
```

Comment:
naming nit: `runUSI` if you insist calling it `USI` or just `run` or `runScheduler`